src/main/scala/com/fulcrumgenomics/alignment/Alignment.scala

```diff
@@ -278,4 +284,93 @@ case class Alignment(query: Array[Byte],
+    * adjusted `queryStart` and `targetStart` positions and an updated `cigar`.  The score is set to 0.
+    *
+    * @param start the 1-based inclusive position of the first base on the target sequence to include
+    * @param end   the 1-based inclusive position of the last base on the target sequence to include
+    * @return a new [[Alignment]] with updated coordinates and cigar
+    */
+  def subByTarget(start: Int, end: Int): Alignment = {
+    require(start >= targetStart && start <= targetEnd, "start is outside of aligned region of target sequence")
+    require(end   >= targetStart && end   <= targetEnd, "end is outside of aligned region of target sequence")
+    sub(start, end, this.targetStart, _.operator.consumesReferenceBases())
+  }
+
+  /**
+    * Private helper method that helps generate an Alignment that is a subset of the current alignment.
+    *
+    * @param start the start (on either the query or target sequence) of the desired window
+    * @param end the end (on either the query or target sequence) of the desired window
+    * @param initialStart the start position on the relevant sequence (either target OR query)
+    * @param consumes a function that returns true if the cigar element passed as a parameter consumes
+    *                 bases on the sequence on which `start` and `end` are expresssed
+    * @return a new [[Alignment]] with adjusted start, end and cigar, and with score=0
+    */
+  private def sub(start: Int, end: Int, initialStart: Int, consumes: CigarElem => Boolean): Alignment = {
+    val elems = new ArrayBuffer[CigarElem](cigar.size)
+    var (qStart, tStart, currStart) = (queryStart,targetStart,initialStart)
+
+    cigar.foreach { elem =>
+      val currEnd = if (consumes(elem)) currStart + elem.length - 1 else currStart
+
+      if (currEnd < start) {
+        // Element before the desired window, need to bump start positions
+        qStart    += elem.lengthOnQuery
+        tStart    += elem.lengthOnTarget
+        if (consumes(elem)) currStart += elem.length
+      }
+      else if (currStart > end) {
+        // Don't include, beyond the range we're interested
+      }
+      else if (currStart >= start && currEnd <= end) {
+        // Contained within the target region; ignore insertions that are at the end of the desired region
+        if (consumes(elem) || (currStart != start && currStart != end)) {
+          elems     += elem
+          if (consumes(elem)) currStart += elem.length
+        }
+      }
+      else {
+        // Element is overlaps the desired region, may enclose, be enclosed by or straddle
+        var len = elem.length
+
+        if (currStart < start) {
+          // Element is split over the start of the desired region
+          val diff = start - currStart
+          len -= diff
+          if (elem.operator.consumesReadBases())      qStart += diff
+          if (elem.operator.consumesReferenceBases()) tStart += diff
+          currStart += diff
```

Comment:
I think you could remove line 358 (`len -= diff`) and line (`currStart += diff`) since you add `len` to `currStart` on 370 anyways.

Reply:
If I comment out those two lines I get a whole lot of test failures.  The reason being here that `len` is the length of the element to be added to the cigar, after adjusting for the fact that the current element may start before and/or end after the desired region.  So it has to get adjusted.  And then the qStart/tStart/curreStart all need to get bumped up as we're moving up the alignment.